### PR TITLE
Update backend deps: Django 4, Postgres 15

### PR DESCRIPTION
**Pull request by rjruizes**
This PR upgrades several Python dependencies used by the backend, importantly Django 3 to 4 and Postgres 9 to 15.

The mallard staging environment was previously undocumented, so I added a doc for that, along with instructions on updating production to the new Postgres version. This code is running in mallard now.

### CODE REVIEW

File: envs/docker-common.yml [hunk 1]
```diff
@@ -2,30 +2,33 @@ version: '2.1'
 
 services:
   postgres:
-    image: postgres:9.6.2
+    image: postgres:15.2
```
Comment (AstridKery) [inline envs/docker-common.yml:5]: Getting this error when I try any `docker-compose` command from the `envs/dev` folder

`service "postgres" refers to undefined volume smart_pgdata_15_2: invalid compose project`

**Comment by AstridKery** [comment]
@rjruizes would you mind confirming that changing the first line in the Dockerfile from 

`FROM python:3.8-buster` to `FROM --platform=linux/amd64 python:3.8-buster` does not break for your computer?

File: envs/docker-common.yml [hunk 2]
```diff
@@ -2,30 +2,33 @@ version: '2.1'
...
       - POSTGRES_PASSWORD=my-secret-password
       - POSTGRES_USER=smart
       - POSTGRES_DB=smart
+      - POSTGRES_HOST=postgres
+      - POSTGRES_HOST_AUTH_METHOD=md5
     ports:
       - "${EXTERNAL_POSTGRES_PORT:-5432}:5432"
     volumes:
-      - smart_pgdata:/var/lib/postgresql/data
+      - smart_pgdata_15_2:/var/lib/postgresql/data
```
Comment (AstridKery) [inline envs/docker-common.yml:15]: Getting this error when I try any docker-compose command from the envs/dev folder

`service "postgres" refers to undefined volume smart_pgdata_15_2: invalid compose project`

Comment (AstridKery) [reply envs/docker-common.yml:15]: When I dropped the _15_2 change it built fine.

Comment (rjruizes) [reply envs/docker-common.yml:15]: Fixed. The change to the volume name was driven by the prod migration, where I needed to support multiple pg volumes at once.

**Comment by rjruizes** [comment]
Verified functional with change to platform in FROM tag and Postgres volume name change.

**Comment by AstridKery** [comment]
@peridoteagle I did a pass and it seems to work locally on my machine. I also ran the backend tests just to make sure there weren't any weird things from updating the backend dependencies and didn't spot any errors. Would you mind testing on Mallard?

**Review by AstridKery — APPROVED** [review]
I think this is ready to go

**Pull request merged**